pkg/services/serviceaccounts/manager/service.go

```diff
@@ -18,34 +20,63 @@ var (
 )
 
 type ServiceAccountsService struct {
-	store serviceaccounts.Store
-	cfg   *setting.Cfg
-	log   log.Logger
+	store      serviceaccounts.Store
+	cfg        *setting.Cfg
+	log        log.Logger
+	UsageStats usagestats.Service
 }
 
 func ProvideServiceAccountsService(
 	cfg *setting.Cfg,
 	store *sqlstore.SQLStore,
 	ac accesscontrol.AccessControl,
 	routeRegister routing.RouteRegister,
+	usageStats usagestats.Service,
 ) (*ServiceAccountsService, error) {
 	s := &ServiceAccountsService{
-		cfg:   cfg,
-		store: database.NewServiceAccountsStore(store),
-		log:   log.New("serviceaccounts"),
+		cfg:        cfg,
+		store:      database.NewServiceAccountsStore(store),
+		log:        log.New("serviceaccounts"),
+		UsageStats: usageStats,
 	}
 	if err := ac.DeclareFixedRoles(role); err != nil {
 		return nil, err
 	}
+	s.registerUsageMetrics()
+
 	serviceaccountsAPI := api.NewServiceAccountsAPI(s, ac, routeRegister)
 	serviceaccountsAPI.RegisterAPIEndpoints(cfg)
 	return s, nil
 }
 
-func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
+func (sa *ServiceAccountsService) IsDisabled() error {
 	if !sa.cfg.FeatureToggles["service-accounts"] {
 		sa.log.Debug(ServiceAccountFeatureToggleNotFound)
+		return errors.New(ServiceAccountFeatureToggleNotFound)
+	}
+	return nil
+}
+
+func (sa *ServiceAccountsService) DeleteServiceAccount(ctx context.Context, orgID, serviceAccountID int64) error {
+	err := sa.IsDisabled()
+	if err != nil {
 		return nil
 	}
 	return sa.store.DeleteServiceAccount(ctx, orgID, serviceAccountID)
 }
+
+func (sa *ServiceAccountsService) registerUsageMetrics() {
+	sa.UsageStats.RegisterMetricsFunc(func(context.Context) (map[string]interface{}, error) {
+		return map[string]interface{}{
+			"stats.serviceaccounts.enabled.count": sa.getUsageMetrics(),
+		}, nil
+	})
+}
+
+func (sa *ServiceAccountsService) getUsageMetrics() interface{} {
+	err := sa.IsDisabled()
+	if err != nil {
+		return 0
+	}
+	return 1
+}
```

Comment:
this is exposed to us! this is what want to get for now

Reply:
We could also count the number of accounts added, deleted, etc